django/db/models/query.py

```diff
@@ -885,18 +886,28 @@ def all(self):
         """
         return self._chain()
 
+    def _not_support_combined_queries(self, name):
+        if self.query.combinator:
+            raise NotSupportedError(
+                'Calling %s() is not supported after %s().' % (name, self.query.combinator)
+            )
+
     def filter(self, *args, **kwargs):
         """
         Return a new QuerySet instance with the args ANDed to the existing
         set.
         """
+        # Ignore check if filter called by .get()
+        if inspect.stack()[1].function != 'get':
+            self._not_support_combined_queries('filter')
```

Comment:
It's a bit :scream: to me. `filter()` doesn't change anything for combined queries so I think we should handle this in `get()`, e.g.
```python
def get(self, *args, **kwargs):
    clone = self._chain() if self.query.combinator else self.filter(*args, **kwargs)
```

Reply:
Thanks